scipy/interpolate/predict.py

```diff
@@ -0,0 +1,114 @@
+""" This module implements the Always Stable Projector from Ref. [1] that can
+be used for predicting values in a (time) series, such as in [2-3].
+
+[1] J. Kolafa, J. Comput. Chem. 25, 335 (2004).
+[2] T. D. Kuehne, M. Krack, F. Mohamed, and M. Parrinello,
+    Phys. Rev. Lett. 98, 066401 (2007).
+[3] T. Spura, H. Elgabarty, and T. D. Kuehne, Phys. Chem. Chem. Phys. (2015).
+"""
+
+import numpy as np
+import scipy.misc as scimisc
+
+__all__ = ["asp", "Cache"]
+
+
+def asp(array):
+    """
+    Run the Always Stable Projector algorithm.
+
+    Parameters
+    ----------
+    array : array_like
+        The observable array that should be projected to the next time step.
+
+    Returns
+    -------
+    The estimated value following the existing values in array.
+    """
+    ret = np.zeros_like(array[0])
+
+    k = len(array)
+    for m in range(1, k+1):
+        n_m = k-m
+        prefactor = (-1.0)**(m+1) * m * scimisc.comb(2*k, k-m) / scimisc.comb(2*k-2, k-1)
+        ret += prefactor*array[n_m]
+    return ret
+
+
+class Cache(object):
```

Comment:
``Cache`` is a too generic name. Also, I don't see a reason why it has to be public.